src/plugins/data/public/ui/query_editor/query_editor_top_row.tsx

```diff
@@ -225,7 +225,17 @@ export default function QueryEditorTopRow(props: QueryEditorTopRowProps) {
   }
 
   function shouldRenderDatePicker(): boolean {
-    return Boolean(props.showDatePicker ?? true) ?? (props.showAutoRefreshOnly && true);
+    return (
+      Boolean(props.showDatePicker ?? true) &&
+      !(
+        queryLanguage &&
+        data.query.queryString.getLanguageService().getLanguage(queryLanguage)?.hideDatePicker
```

Comment:
should we have updating cypress tests? or write some here

Reply:
+1. A unit test too atleast